### PR TITLE
initial implementation FalkorDBPropertyGraphStore

### PR DESCRIPTION
**Pull request by AviAvni**
# Description

Implement PropertyGraphStore for falkordb

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by logan-markewich — COMMENTED** [review]
Very cool! Thanks a ton for contributing this.

I think overall the code seems fine, but I want to try and test this locally today. Will merge if it looks good :)

**Comment by logan-markewich** [comment]
`docker pull falkordb:latest` gives me an error about it not existing or requiring permissions 👀

**Comment by logan-markewich** [comment]
Ah, looks like it should be `falkordb/falkordb:latest`

File: docs/docs/examples/property_graph/property_graph_falkordb.ipynb [hunk 1]
```diff
@@ -0,0 +1,345 @@
+{
```
Comment (logan-markewich) [inline docs/docs/examples/property_graph/property_graph_falkordb.ipynb:1]: <p>What's the default username and password? I wasn't able to login</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/14936/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (logan-markewich) [inline docs/docs/examples/property_graph/property_graph_falkordb.ipynb:1]: <p>I think this block is specific to neo4j, we should update it</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/14936/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (logan-markewich) [inline docs/docs/examples/property_graph/property_graph_falkordb.ipynb:1]: <p>I got the following error on this block</p><p> </p>

<pre class="ql-syntax" spellcheck="false">ResponseError: Error running coroutine: LIMIT specified value of invalid type, must be a positive integer
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/14936/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (logan-markewich) [inline docs/docs/examples/property_graph/property_graph_falkordb.ipynb:1]: <p>Due to the above error, I couldn't run the query engine</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/14936/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Comment by logan-markewich** [comment]
Lets make sure the notebook works, then we can merge :)

File: docs/docs/examples/property_graph/property_graph_falkordb.ipynb (hunk 1 above)
Comment (AviAvni) [reply docs/docs/examples/property_graph/property_graph_falkordb.ipynb:1]: empty

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**